g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationWithPatternAction.java

```diff
@@ -51,7 +53,14 @@
 	    */
 	   public APIResponseDTO getTransPattern(TranslationWithPatternDTO data) throws Exception {
 	      logger.info("begin getTransPattern");
-	      if (validate(data)) {
+	      boolean validateResult;
+	      try {
+	    	  validateResult = validate(data);
+	      }catch(VIPAPIException excep) {
+	    	  return super.handleResponse(APIResponseStatus.BAD_REQUEST.getCode(),
+	    			  excep.getMessage(), null);
+	      }
+	      if (validateResult) {
```

Comment:
How about change validate throwing VIPAPIException for other conditions to unify the process?

Reply:
I agree with you, But need to change a lot other business code and this API will delete after JS client change to get API